src/web_scraper/web_scraper.py

```diff
@@ -0,0 +1,131 @@
+"""
+Description: this program will eventually be responsible for extracting RSS senate/house bill metadata and DOCX full bill, and then uploading to supabase table
+"""
+
+from docx import Document
+from supabase import create_client, Client
+from datetime import datetime
+import feedparser
+import os
+import re
+
+from dotenv import load_dotenv
+# load_dotenv() #for local secret management with .env file
+
+senate_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/SenateBills.xml"
+house_rss_feed = "https://www.legis.state.pa.us/WU01/LI/RSS/HouseBills.xml"
+
+sb_api_url = "https://vsumrxhpkzegrktbtcui.supabase.co"
+#sb_api_key = os.getenv("SUPABASE_API_KEY") #local secret management wtih .env file
+sb_api_key = os.environ["SUPABASE_API_KEY"] #github actions secret management 
+
+def get_bill_no_from_RSS_title(title, leg_body):
+    senate_bill_no_regex = '(Senate Bill \d*|Senate Resolution \d*)'
+    house_bill_no_regex = '(House Bill \d*|House Resolution \d*)'
+    
+    if leg_body == "senate": return re.findall(senate_bill_no_regex, title)[0].split(' ')[-1]
+    if leg_body == "house": return re.findall(house_bill_no_regex, title)[0].split(' ')[-1]
+    return title #return the original title if legislative body not specified
+    
+def get_bill_printer_no_from_RSS_title(title):
+    printer_no_regex="Printer's Number \d*"
+    return re.findall(printer_no_regex, title)[0].split(' ')[-1]
+
+def check_for_existing_bill_supabase_records(supa_con, bill_no, printer_no, leg_body):
+    existing_bill_records = supa_con.table('Revisions').select("*", count="exact").eq("bill_id", bill_no).eq("printer_no", printer_no).eq("legislative_body", leg_body).execute()
+    if existing_bill_records.count > 0: return True
+    else: return False
+
+def extract_and_upload_senate_bill_metadata(supa_con):
+    sen_rss = feedparser.parse(senate_rss_feed)
+    for bill in sen_rss.entries:
+        if bill is None: continue
+
+        bill_no = get_bill_no_from_RSS_title(bill.title, "senate")
+        printer_no = get_bill_printer_no_from_RSS_title(bill.title)
+        
+        if check_for_existing_bill_supabase_records(supa_con, bill_no, printer_no, "senate"): continue
```

Comment:
A minor comment that I won't hold back the PR on, but in the future, having it be something like "has_existing_supabase_records" would be more informative -- "check_for_existing_bill_supabase_records" does seem to imply that it returns true if there are existing supabase records, but it's a tad more ambiguous.